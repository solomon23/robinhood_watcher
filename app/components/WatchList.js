// @flow
import React, { Component } from 'react'
import WatchRow from './WatchRow'
import styles from './styles/StockList.scss'

type Props = {
  stocks: Array<WatchStock>
}

export default class WatchList extends Component<Props> {
  props: Props

  render() {
    const { stocks } = this.props

    if (!stocks || stocks.length === 0) {
      return null
    }

    return (
      <div className={styles.stockList}>
        {stocks.sort((a, b) => a.symbol.localeCompare(b.symbol)).map(stock => (
          <WatchRow key={stock.symbol} stock={stock} />
        ))}
      </div>
    )
  }
}
