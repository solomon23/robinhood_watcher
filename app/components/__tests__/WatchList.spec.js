import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { shallowToJson } from 'enzyme-to-json'
import WatchList from '../WatchList'

Enzyme.configure({ adapter: new Adapter() })

let actions = null

const defaultState = {
  stocks: [],
}

describe('WatchList component', () => {
  beforeEach(() => {
    actions = { }
  })

  it('should display empty', () => {
    const component = shallow(<WatchList {...defaultState} {...actions} />)
    expect(shallowToJson(component)).toMatchSnapshot()
  })

  it('should sort list', () => {
    const state = { ...defaultState, stocks: [{ symbol: 'z' }, { symbol: 'a' }, { symbol: 'c' }] }
    const component = shallow(<WatchList {...state} {...actions} />)
    expect(shallowToJson(component)).toMatchSnapshot()
  })

  it('should show as watchlist', () => {
    const state = { ...defaultState, stocks: [{ symbol: 'z' }, { symbol: 'a' }, { symbol: 'c' }] }
    const component = shallow(<WatchList {...state} {...actions} watchlist />)
    expect(shallowToJson(component)).toMatchSnapshot()
  })
})
