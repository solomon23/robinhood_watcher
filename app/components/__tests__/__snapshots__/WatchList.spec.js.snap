// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WatchList component should display empty 1`] = `""`;

exports[`WatchList component should show as watchlist 1`] = `
<div
  className="stockList"
>
  <WatchRow
    key="a"
    stock={
      Object {
        "symbol": "a",
      }
    }
  />
  <WatchRow
    key="c"
    stock={
      Object {
        "symbol": "c",
      }
    }
  />
  <WatchRow
    key="z"
    stock={
      Object {
        "symbol": "z",
      }
    }
  />
</div>
`;

exports[`WatchList component should sort list 1`] = `
<div
  className="stockList"
>
  <WatchRow
    key="a"
    stock={
      Object {
        "symbol": "a",
      }
    }
  />
  <WatchRow
    key="c"
    stock={
      Object {
        "symbol": "c",
      }
    }
  />
  <WatchRow
    key="z"
    stock={
      Object {
        "symbol": "z",
      }
    }
  />
</div>
`;
